File: euler_totient/Python/euler_totient.py
# Python 3 program to calculate
# Euler's Totient Function
# using Euler's product formula

def run(n) :

result = n # Initialize result as n

# Consider all prime factors
# of n and for every prime
# factor p, multiply result with (1 – 1/p)
p=2
while(p*p<=n) :

# Check if p is a prime factor.
if (n % p == 0) :

# If yes, then update n and result
while (n % p == 0) :
n = n // p
result = result * (1.0 – (1.0 / (float) (p)))
p = p + 1

# If n has a prime factor
# greater than sqrt(n)
# (There can be at-most one
# such prime factor)
if (n > 1) :
result = result * (1.0 – (1.0 / (float)(n)))

return (int)(result)

# Driver program to test above function
for n in range(1,11) :
print(“run(“,n,”) = “,run(n))

File: euler_totient/Cpp/euler_totient.cpp
#include <bits/stdc++.h>

using namespace std;

int phi(int n) {
    int result = n;
    for (int i = 2; i * i <= n; i++) {
        if(n % i == 0) {
            while(n % i == 0)
                n /= i;
            result -= result / i;
        }
    }
    if(n > 1)
        result -= result / n;
    return result;
}

int main(){

   cout<<"Euler totient function from 10 to 14 is : "<<endl;
   cout<<phi(10)<<endl;
   cout<<phi(11)<<endl;
   cout<<phi(12)<<endl;
   cout<<phi(13)<<endl;
   cout<<phi(14)<<endl;

   return 0;
}